master/internal/api_experiment.go

```diff
@@ -0,0 +1,45 @@
+package internal
+
+import (
+	"context"
+	"strings"
+
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+func (a *apiServer) GetExperiments(
```

Comment:
praise: this looks super clean! Nice pattern for the rest of the list endpoints to follow.